### PR TITLE
Added an option to the DEFINE statement to choose options without replacement

### PR DESCRIPTION
**Pull request by ThaumicMekanism**
This allows you to specify different targets you want to replace with the same set of values. Example usage:

```
# DEFINE MATCH [x y] [a b x y]
```

This will set x to either a, b, x, or y then set y to any of the options which are left.

### CODE REVIEW

File: examtool/api/convert.py [hunk 1]
```diff
@@ -108,11 +110,29 @@ def consume_rest_of_question(buff):
                     **parse("\n".join(contents)),
                     "options": options,
                     "substitutions": substitutions,
+                    "substitutions_match": substitutions_match
                 }
             else:
                 raise SyntaxError("Unexpected END in QUESTION")
         elif mode == "DEFINE":
-            substitutions[directive] = rest.split(" ")
+            if directive == "MATCH":
```
Comment (rahularya50) [inline examtool/api/convert.py]: Please extract into a function rather than duplicating logic

File: examtool/api/convert.py [hunk 2]
```diff
@@ -215,6 +272,7 @@ def convert(text):
         "groups": groups,
         "config": config,
         "substitutions": substitutions,
+        "substitutions_match": substitutions_match
```
Comment (rahularya50) [inline examtool/api/convert.py]: nit: trailing comma

File: examtool/api/scramble.py [hunk 3]
```diff
@@ -6,6 +6,7 @@ def scramble(email, exam, *, keep_data=False):
 
     def scramble_group(group, substitutions, config, depth):
         group_substitutions = select(group["substitutions"])
+        group_substitutions.update(select_no_replace(group["substitutions_match"] if "substitutions_match" in group else []))
```
Comment (rahularya50) [inline examtool/api/scramble.py]: `group.get("substitutions_match", [])`, please

File: examtool/api/scramble.py [hunk 4]
```diff
@@ -85,6 +88,20 @@ def get_elements(group):
...
         out[k] = random.choice(v)
     return out
+
+def select_no_replace(substitutions_match):
+    out = {}
+    # DEFINE MATCH
+    for item in substitutions_match:
+        k = item["directives"]
+        v = item["replacements"]
+        values = v.copy()
```
Comment (rahularya50) [inline examtool/api/scramble.py:102]: Why do you `.copy()`? `scramble` is destructive anyway.

**Review by rahularya50 — CHANGES_REQUESTED** [review]

**Comment by rahularya50** [comment]
TODO: Update docs in `examtool-web` (@rahularya50)

File: examtool/api/convert.py (hunk 1 above)
Comment (ThaumicMekanism) [reply examtool/api/convert.py]: Yup good point. Was gonna do that but had forgot.

File: examtool/api/convert.py (hunk 2 above)
Comment (ThaumicMekanism) [reply examtool/api/convert.py]: forgot to add those back.

File: examtool/api/scramble.py (hunk 3 above)
Comment (ThaumicMekanism) [reply examtool/api/scramble.py]: Rip forgot about that.

File: examtool/api/scramble.py (hunk 4 above)
Comment (ThaumicMekanism) [reply examtool/api/scramble.py:102]: I think it was because I didn't want to directly modify the other list just incase it is used elsewhere.

**Pull request merged**